pkg/controller/podautoscaler/horizontal.go

```diff
@@ -425,6 +429,13 @@ func (a *HorizontalController) computeReplicasForMetric(ctx context.Context, hpa
 			return 0, "", time.Time{}, condition, fmt.Errorf("failed to get %s resource metric value: %v", spec.Resource.Name, err)
 		}
 	case autoscalingv2.ContainerResourceMetricSourceType:
+		if !a.containerResourceMetricsEnabled {
+			// If the container resource metrics feature is disabled but the object has the one,
+			// that means the user enabled the feature once, created some HPAs with the container resource metrics, and disabled it finally.
+			// We cannot return errors in this case because that'll result in all HPAs with the container resource metric sources failing to scale down.
+			// Thus, here we silently ignore it and return current replica values so that it won't affect the autoscaling decision
+			return specReplicas, "", time.Time{}, condition, nil
```

Comment:
If you always return current replica size from one of the metrics you will effectively also prevent scale down, right? The test could simulate scale down on another metric to verify this scenario.

Reply:
re 1) I would prefer container metrics to always fail with a clear message about feature disabled than silently passing with current size. The effect would be the same – no scaling down, scale up possible if other metric recommends it – but at least it's clear the situation is not healthy. Silently passing current size seems to me like a recipe for surprises, especially if feature gate is later enabled and HPA suddenly recommends scale downs.

re 2) IMO the way to implement it would be to skip the ContainerResource metric spec when iterating to compute metric recommendations and put some condition informing about the fact.

I don't see how (2) makes the situation better for cluster admins. They still should coordinate with HPA owners, because HPA behavior is going to be different than owner's intent, with a risk of unexpected scale down from a different metric. In both (1) and (2) the code path for Container Resource is not called, so no difference from the controller health PoV (performance, memory leaks, whatever). So I would choose (1) (but with error instead of silent passing) as safer for workloads health and clear for HPA owners.

Reply:
Totally make sense. Let's go with (1).
Let me:
- change this line to return the error (like "The container resource metrics is disabled")
- add the test case to verify the scenario.

Reply:
Done https://github.com/kubernetes/kubernetes/pull/116043/commits/f76258f0ff972942ab6722043b947e58f29e66e2